ext/updater.go

```diff
@@ -287,6 +259,33 @@ func (u *Updater) Stop() error {
 	return nil
 }
 
+func (u *Updater) StopBot(b *gotgbot.Bot) bool {
+	bData, ok := u.botMapping.removeBot(b)
+	if !ok {
+		return false
+	}
+
+	stopBot(bData)
+	return true
+}
+
+func (u *Updater) StopAllBots() {
+	for _, bData := range u.botMapping.removeAllBots() {
+		stopBot(bData)
+	}
+}
+
+func stopBot(data botData) {
```

Comment:
Conceptually, this could be a method on botData?
```suggestion
func (data botData) stop() {
```

Reply:
Also, probably it would make sense to have "stopped atomic.Bool" on the `botData` structure, and write "true" into it on "stop()".  This would prevent panic, should stop be called more than once (by mistake).